pkg/server/internal/packetio.go

```diff
@@ -150,19 +150,21 @@ func (p *PacketIO) readOnePacket() ([]byte, error) {
 		if _, err := io.ReadFull(r, header[:]); err != nil {
 			return nil, errors.Trace(err)
 		}
+
+		// To be compatible with MariaDB Java Connecter 2.X, 
+		// check sequence only for the packet with compression protocol disabled.
+		sequence := header[3]
+		if sequence != p.sequence {
+			return nil, server_err.ErrInvalidSequence.GenWithStack(
+				"invalid sequence, received %d while expecting %d", sequence, p.sequence)
+		}
 	} else {
 		if _, err := io.ReadFull(p.compressedReader, header[:]); err != nil {
 			return nil, errors.Trace(err)
 		}
 	}
 
 	length := int(uint32(header[0]) | uint32(header[1])<<8 | uint32(header[2])<<16)
-	sequence := header[3]
-
-	if sequence != p.sequence {
-		return nil, server_err.ErrInvalidSequence.GenWithStack(
```

Comment:
Maybe we could log an error instead of returning? Then we can still keep this runtime check.

Reply:
> Maybe we could log an error instead of returning? Then we can still keep this runtime check.

Yeah. For the normal packet, the sequence check is supposed to be passed. As for the compressed packet, we generate a log and then let the process continue. Did I get that right?

Reply:
Yes indeed.